web-registry/src/hooks/useQueryBalances.ts

```diff
@@ -0,0 +1,60 @@
+import { useCallback, useEffect, useState } from 'react';
+import {
+  DeepPartial,
+  QueryBalancesRequest,
+  QueryBalancesResponse,
+  QueryClientImpl,
+} from '@regen-network/api/lib/generated/regen/ecocredit/v1/query';
+
+import { useLedger } from '../ledger';
+
+type FetchBalances = (
+  request: DeepPartial<QueryBalancesRequest>,
+) => Promise<QueryBalancesResponse | undefined>;
+
+type Params = {
+  address?: string;
+};
+
+export default function useQueryBalances({ address }: Params): {
```

Comment:
Need a separate hook here to have a method for refetch in the hook's response.

Reply:
I understand... it's a repeating hook pattern... Sounds good to me at the moment. A next step to avoid this kind of hooks could be to extend the base hooks (eg `useEcocreditQuery`) and add lazy fetch functionality, what do you think?

Reply:
I agree it would be best to evolve `useEcocreditQuery` for that. Adding a note to create an issue for that.